easy-ui-react/src/Notification/Notification.tsx

```diff
@@ -0,0 +1,344 @@
+ * showSuccessToast(content: ToastProps): void;
+ * showWarningToast(content: ToastProps): void;
+ * showErrorToast(content: ToastProps): void;
+ * showPromotionalAlert(content: AlertProps): void;
+ * showNeutralAlert(content: AlertProps): void;
+ * showSuccessAlert(content: AlertProps): void;
+ * showWarningAlert(content: AlertProps): void;
+ * showErrorAlert(content: AlertProps): void;
+ * 
+ * type ToastProps = {
+ *   // Notification message
+ *   message: ReactNode;
+ *   // Notification will render with left aligned status icon
+ *   hasIcon?: boolean;
+ * }
+ * 
+ * type AlertProps = {
+ *   // Notification message
+ *   message: ReactNode;
+ *   // Notification will render with left aligned status icon
+ *   hasIcon?: boolean;
+ *   // Callback function when dismissing alerts
+ *   onDismiss?: () => void;
+ * }
+ * ```
+ * @example
+ * _Success:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*  const onToast = () => {
+*    notification.showSuccessToast({ message: "message" });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showSuccessAlert({ message: "message" });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _Without status icons:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*  const onToast = () => {
+*    notification.showPromotionalToast({ message: "message", hasIcon: false });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showWarningAlert({ message: "message", hasIcon: false });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _With onDismiss callback, only applicable for alerts:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*
+*  const onAlert = () => {
+*    notification.showNeutralAlert({
+*      message: "message",
+*      onDismiss: () => alert("i am called immediately before the alert closes"),
+*    });
+*  };
+*
+*  return (
+*    <>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+ */
+export function useNotification() {
+  const notification = useContext(NotificationContext);
+  if (!notification) {
+    throw new Error(
+      "useNotification() must be used within a NotificationProvider, which is bundled with EasyUIProvider",
+    );
+  }
+  return { notification: notification };
```

Comment:
**question:** would it make sense to expose this as just `return notification;` instead of as an intermediary object `return { notification };`?

```tsx
const notification = useNotification();
const onToast = () => { notification.showToast(); }
```

or are you designing it to be extensible in the future?

Reply:
agree, it does make sense to expose it just as notification instead of an intermediary object.